index.js

```diff
@@ -71,7 +72,25 @@ const onMergedStreamEnd = async (passThroughStream, {signal}) => {
 	} catch {}
 };
 
+const onInputStreamsUnpipe = async (streams, passThroughStream, {signal}) => {
+	const streamsSet = new Set(streams);
+	for await (const [stream] of on(passThroughStream, 'unpipe', {signal})) {
+		if (streamsSet.has(stream)) {
+			stream.emit(unpipeEvent);
+		}
+	}
+};
```

Comment:
Unfortunately, the `unpipe` event is emitted only on the destination, not the source, so we need to do this.